### PR TITLE
Implement integrations specs and fix n+1 problems

### PR DESCRIPTION
**Pull request by amentetteh**
In this PR, we followed below requirements:

- Solve the N+1 problem when fetching all posts and their comments for a user. Screenshots of the results are as follows: 

- Use Capybara to write integration tests for each view in our project, which includes user index page, user show page, post index page, and post show page

### CODE REVIEW

**Comment by DammyShittu** [comment]
Hi @amentetteh @sebalp100 and @SimonGideon,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a project that has failed linter checks

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/integration/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    it 'should render name of commenters' do
+      expect(page).to have_content(@user1.name)
+      expect(page).to have_content(@user2.name)
+    end
+
+    it 'should render content of comments' do
+      expect(page).to have_content(@first_comment.text)
+      expect(page).to have_content(@second_comment.text)
+    end
+  end
```
Comment (DammyShittu) [inline spec/integration/posts_spec.rb]: - **[OPTIONAL]** We want to organize the tests. Kindly move this post show tests to their own file. We are testing 4 different views, so it makes sense that we should have 4 different corresponding test files.

File: spec/integration/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      expect(page).to have_content(@third_post.text)
+    end
+    it "should have a button to view all of a user's posts" do
+      expect(page).to have_link('See all posts', href: user_posts_path(user_id: @user.id))
+    end
+    it 'should redirects to all posts show page.' do
+      click_link 'See all posts'
+      expect(page).to have_current_path(user_posts_path(user_id: @user.id))
+    end
+  end
```
Comment (DammyShittu) [inline spec/integration/users_spec.rb]: - **[OPTIONAL]** Kindly move these tests to their own file.

Comment (DammyShittu) [inline spec/integration/users_spec.rb]: - [ ] For the user show page, kindly add:

  - Test that redirects to the post's show page when you click on a user's post.

File: spec/integration/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    it 'should render the pagenation button' do
+      expect(page).to have_content('Pagination')
+    end
+
+    it "should redirect to that post's show page" do
+      click_link @post.text
+      expect(page).to have_current_path(user_post_path(user_id: @user.id, id: @post.id))
+    end
+  end
```
Comment (DammyShittu) [inline spec/integration/posts_spec.rb]: - [ ] For the post index page, kindly add:

  - Test that asserts you can see a post's title.

File: spec/integration/posts_spec.rb (hunk 1 above)
Comment (DammyShittu) [inline spec/integration/posts_spec.rb]: - [ ] For the post show page, kindly add:

  - Test that asserts you can see who wrote the post.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @amentetteh, @sebalp100 and @SimonGideon,

## Highlights
✔️ Capybara for tests

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integration/posts_spec.rb (hunk 1 above)
Comment (amentetteh) [reply spec/integration/posts_spec.rb]: Thanks.

**Review by DammyShittu — APPROVED** [review]
Hi @amentetteh, @sebalp100 and @SimonGideon,

## APPROVED 🟢

## Highlights
✔️ Tests are well organized

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**